workshop/exercise-2/README.md

```diff
@@ -14,21 +14,52 @@ The application for this lab is a simple node.js with express app which returns
 	cd fib-knative
 	```
 
+## Set up a Private Container Registry and Obtain Credentials
+In this lab, we'll use the IBM Container Registry to host our docker images since you already have access to this through your IBM Cloud Account. IBM Container Registry enables you to store and distribute container images in a fully managed private registry.
+
+1. Install the container registry plugin:
+
+  ```
+  ibmcloud plugin install container-registry
+  ```
+2. Choose a name for your first namespace, and create it. 
+
+  ```
+  ibmcloud cr namespace-add <my_namespace>
+  ```
+3. Create a token. This token is a non-expiring token with read and write access to all namespaces the region.
+
+  ```
+  ibmcloud cr token-add --description "read write token for all namespaces" --non-expiring --readwrite
+  ```
+
+4. The CLI output should include a Token Identifier and the Token. Make note of the Token for later in this lab. You will not need the Token Identifier. You can verify that the token was created by listing all tokens.
+
+  ```
+  ibmcloud cr token-list
+  ```
 
 ### Provide Container Registry Credentials
-This lab will need credentials for authenticating to your container registry - we'll be using dockerhub. You could also use the IBM Container Registry. First, we'll need to create a `Secret` to store the credentials.
+This lab will need credentials for authenticating to your private container registry. First, we'll need to create a `Secret` to store the credentials for this registry. This secret will be used for the knative-serving component to pull down an image from the container registry.
 
 A `Secret` is a Kubernetes object containing sensitive data such as a password, a token, or a key. You can also read more about [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
 
-1. To create this object, we'll first need to base64 encode our username and password for dockerhub.
+1. Let's create a the secret, which will be a `docker-registry` type secret. This type of secret is used to authenticate with a container registry to pull down a private image. We can create this via the commandline.
+
+  ```
+  kubectl create secret docker-registry ibm-cr-secret --docker-server=https://registry.ng.bluemix.net --docker-username=token --docker-password=<token_value>
+  ```
+
+2. We will also need a secret for the kaniko build template to be able to push the built image to the container registry. To create this object, we'll first need to base64 encode our username and password for IBM Container Registry. For username, simply use the string `token`. The base64 encoding of `token` should be: `dG9rZW4=`.  For token_value, use the token you made note of earlier.
```

Comment:
I like that you mentioned that the username is `token` but let's move that to the previous step since that's the first spot where we actually do that (under docker-username).